src/common/libkvs/kvs_dir.c

```diff
@@ -106,33 +109,27 @@ flux_kvsdir_t *flux_kvsdir_create (flux_t *handle, const char *rootref,
     return NULL;
 }
 
-const char *flux_kvsdir_tostring (flux_kvsdir_t *dir)
+const char *flux_kvsdir_tostring (const flux_kvsdir_t *dir)
 {
-    if (!dir->dirobj_string) {
-        if (!(dir->dirobj_string = json_dumps (dir->dirobj, JSON_COMPACT))) {
-            errno = ENOMEM;
-            return NULL;
-        }
-    }
-    return dir->dirobj_string;
```

Comment:
should the change from using ```dirobj_string``` to ```json_str``` be in a different commit, b/c this doesn't seem to be mapped to this commit message.  It appears this also fixes a mem-leak.

Reply:
Yes that looks like a rebase error.  I'll fix...